Meadowlark-js/backends/meadowlark-mongodb-backend/test/middleware/SecurityMiddleware.test.ts

```diff
@@ -0,0 +1,313 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { FrontendRequest, FrontendResponse, newFrontendRequest, newFrontendResponse } from '@edfi/meadowlark-core';
+import { newFrontendRequestMiddleware } from '@edfi/meadowlark-core/src/handler/FrontendRequest';
+import { newPathComponents } from '@edfi/meadowlark-core/src/model/PathComponents';
+import { securityMiddleware } from '../../src/security/SecurityMiddleware';
+import * as OwnershipSecurity from '../../src/repository/OwnershipSecurity';
+
+jest.setTimeout(40000);
+
+describe('given the upsert where response already posted', () => {
+  let result;
+  const mongoClienttMock = {};
+
+  const frontendRequest: FrontendRequest = {
+    ...newFrontendRequest(),
+    action: 'upsert',
+    middleware: {
+      ...newFrontendRequestMiddleware(),
+      pathComponents: { ...newPathComponents() },
+    },
+  };
+
+  const frontendResponse: FrontendResponse = {
+    ...newFrontendResponse(),
+    statusCode: 200,
+  };
+
+  beforeAll(async () => {
+    result = await securityMiddleware({ frontendRequest, frontendResponse }, mongoClienttMock as any);
+  });
+
+  it('should not do anything', async () => {
+    expect(result.frontendResponse.statusCode).toBe(200);
+
+    expect(result.frontendRequest).toEqual({
+      ...newFrontendRequest(),
+      action: 'upsert',
+      middleware: {
+        ...newFrontendRequestMiddleware(),
+        pathComponents: { ...newPathComponents() },
+      },
+    });
+  });
+});
+
+describe('given the upsert where authorization strategy type is different than OWNERSHIP_BASED', () => {
+  let result;
+  const mongoClienttMock = {};
+  const frontendResponse: any = null;
+
+  const frontendRequest: FrontendRequest = {
+    ...newFrontendRequest(),
+    action: 'upsert',
+    middleware: {
+      ...newFrontendRequestMiddleware(),
+      pathComponents: { ...newPathComponents() },
+    },
+  };
+
+  beforeAll(async () => {
+    result = await securityMiddleware({ frontendRequest, frontendResponse }, mongoClienttMock as any);
+  });
+
+  it('should not do anything', async () => {
+    expect(result.frontendResponse).toBe(null);
+
+    expect(result.frontendRequest).toEqual({
+      ...newFrontendRequest(),
+      action: 'upsert',
+      middleware: {
+        ...newFrontendRequestMiddleware(),
+        pathComponents: { ...newPathComponents() },
+      },
+    });
+  });
+});
+
+describe('given the upsert where authorization strategy type is equal to OWNERSHIP_BASED', () => {
```

Comment:
Is this a duplicate of the next test, on line 127? Both have `OWNERSHIP_BASED` and `ACCESS_APPROVED`.